core/src/main/java/org/apache/accumulo/core/logging/ScanUserDataLogger.java

```diff
@@ -0,0 +1,169 @@
+
+public class ScanUserDataLogger extends AbstractLogger {
+
+  private static final long serialVersionUID = 1L;
+  private static final Logger LOG = LoggerFactory.getLogger(ScanUserDataLogger.class);
+  private static final ThreadLocal<String> USER_DATA = new ThreadLocal<>();
+
+  public static void logTrace(Logger classLogger, String userData, String msg, Object... objects) {
+    if (classLogger != null) {
+      classLogger.trace(msg, objects);
+    }
+    if (!LOG.isTraceEnabled()) {
+      return;
+    }
+    USER_DATA.set(userData);
+    LOG.trace(msg, objects);
+  }
+
+  public static void logDebug(Logger classLogger, String userData, String msg, Object... objects) {
+    if (classLogger != null) {
+      classLogger.debug(msg, objects);
+    }
+    if (!LOG.isDebugEnabled()) {
+      return;
+    }
+    USER_DATA.set(userData);
+    LOG.debug(msg, objects);
+  }
+
+  public static void logInfo(Logger classLogger, String userData, String msg, Object... objects) {
+    if (classLogger != null) {
+      classLogger.info(msg, objects);
+    }
+    if (!LOG.isInfoEnabled()) {
+      return;
+    }
+    USER_DATA.set(userData);
+    LOG.info(msg, objects);
+  }
+
+  public static void logWarn(Logger classLogger, String userData, String msg, Object... objects) {
+    if (classLogger != null) {
+      classLogger.warn(msg, objects);
+    }
+    if (!LOG.isWarnEnabled()) {
+      return;
+    }
+    USER_DATA.set(userData);
+    LOG.warn(msg, objects);
+  }
+
+  public static void logError(Logger classLogger, String userData, String msg, Object... objects) {
+    if (classLogger != null) {
+      classLogger.error(msg, objects);
+    }
+    USER_DATA.set(userData);
+    LOG.error(msg, objects);
+  }
+
+  @Override
+  public boolean isTraceEnabled() {
+    return LOG.isTraceEnabled();
+  }
+
+  @Override
+  public boolean isTraceEnabled(Marker marker) {
+    return LOG.isTraceEnabled(marker);
+  }
+
+  @Override
+  public boolean isDebugEnabled() {
+    return LOG.isDebugEnabled();
+  }
+
+  @Override
+  public boolean isDebugEnabled(Marker marker) {
+    return LOG.isDebugEnabled(marker);
+  }
+
+  @Override
+  public boolean isInfoEnabled() {
+    return LOG.isInfoEnabled();
+  }
+
+  @Override
+  public boolean isInfoEnabled(Marker marker) {
+    return LOG.isInfoEnabled(marker);
+  }
+
+  @Override
+  public boolean isWarnEnabled() {
+    return LOG.isWarnEnabled();
+  }
+
+  @Override
+  public boolean isWarnEnabled(Marker marker) {
+    return LOG.isWarnEnabled(marker);
+  }
+
+  @Override
+  public boolean isErrorEnabled() {
+    return LOG.isErrorEnabled();
+  }
+
+  @Override
+  public boolean isErrorEnabled(Marker marker) {
+    return LOG.isErrorEnabled(marker);
+  }
+
+  @Override
+  protected String getFullyQualifiedCallerName() {
+    return null;
+  }
+
+  @Override
+  @SuppressFBWarnings(value = "RV_RETURN_VALUE_IGNORED",
+      justification = "Return value is self, using fluent object in non-fluent manner")
```

Comment:
I'm assuming the warning being suppressed here is the builder, whose methods return `this`.

This could be worded slightly better:
```suggestion
  @SuppressFBWarnings(value = "RV_RETURN_VALUE_IGNORED",
      justification = "fluent builder methods return itself, which we don't need to use")
```

But, given that the implementation could change to have immutable intermediates rather than return `this`, it would probably be better to just use the builder API as intended.